golden: debian 11 efi-based packer implementation

sdake opened the pull request.

rstarmer commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,22 +1,18 @@
-d-i netcfg/get_domain string local
-d-i anna/choose_modules string network-console
+d-i netcfg/get_domain string artificialwisdom.cloud
+d-i netcfg/wireless_wep string
```
do we need WEP config? I'd be surprised if we have a WIFI adapter available?

sdake replied:
web config is not needed. maybe ages ago, and nobody has updated the docs. Nobody including me :)

rstarmer reviewed (changes requested):
@sdake This system still builds a BIOS based image.  I believe we were trying for an EFI build, which this doesn't appear to achieve.

Can you verify that the build generates an UEFI based image?

MostAwesomeDude reviewed (dismissed):
LGTM

I appreciate the README!

sdake commented:
Trying to unwedge github

sdake commented on platform/golden/bullseye.pkr.hcl:
```diff
@@ -97,16 +86,12 @@ source "virtualbox-iso" "base-debian-amd64" {
-# Use Ansible (needs to be on the "packer" host) to provision the instance
   provisioner "ansible" {
     playbook_file = "./provisioners/01_update_packer_user/packer.yml"
-    ansible_env_vars = [
```
ignore this part of the change. I will rebase this change out.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,109 @@
-d-i passwd/user-fullname string Artificial Wisdom
-d-i passwd/user-default-groups string users
-d-i passwd/user-default-groups string sudo
+d-i passwd/user-fullname string ArtificialWisdom
```
full name may have a space.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,109 @@
+## B.4.7.aw1 Boot defaults
 
-d-i grub-installer/only_debian boolean true
-d-i grub-installer/bootdev  string default
```
I will try re-adding line 85.

rstarmer replied:
this is supposed to deal with "whatever" path the /dev/{s,v}d{a,b} device lives on

sdake replied:
For clarity, I did dig into this. The installer, as of `Debian 11.7`, will auto-select a disk, if there is only one when configuring GRUB. The installer uses the `UUID`, not the device name. Booting by `UUID` or by `LABEL` is the modern recommendation. Booting by device names is fraught with problems.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,109 @@
-d-i grub-installer/bootdev  string default
+d-i partman-efi/non_efi_system boolean true
+d-i partman-partitioning/choose_label select gpt
+d-i partman-partitioning/default_label string gpt
```
does gpt require an MBR? I notice the disk has an MBR.

rstarmer replied:
yes, the MBR is where the GPT signature is supposed to be written (IIRC)

sdake replied:
reading the specifications of UEFI, MBR is not required when GPT is specified, with a complaint bootloader. Many past bootloaders were, however, non-compliant and they needed the MBR to recognize the disk.

Current bootloaders on modern hardware are compliant. However, there is no way, that is easy, to disable the writing of the MBR to the disk.

TLDR; the question is not relevant, thought i'd share what I learned.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,109 @@
-#####################
-## Add late commands
 
-d-i preseed/late_command string echo "packer ALL=(ALL:ALL) NOPASSWD:ALL" > /target/etc/sudoers.d/packer && chmod 0440 /target/etc/sudoers.d/packer
```
reinstall this.

sdake replied:
This was relocated to the end of the `preseed.cfg` file.

sdake commented on platform/golden/provisioners/01_update_packer_user/roles/packer-finalize/tasks/main.yml:
```diff
@@ -16,7 +16,7 @@
     state: present
     key: "{{ lookup('url', 'https://github.com/sdake.keys', split_lines=False) }}"
 
-- name: Set authorized keys for packer user - sdake
```
put this in a different pr.

sdake commented on platform/golden/provisioners/01_update_packer_user/roles/packer-finalize/tasks/main.yml:
```diff
@@ -29,11 +29,3 @@
-- name: Update grub2 config to support tty at boot
-  ansible.builtin.lineinfile:
-    path: /etc/default/grub
-    line: 'GRUB_CMDLINE_LINUX="$GRUB_CMDLINE_LINUX console=ttyS0"'
```
add this to boot cmdline.

sdake commented on platform/golden/bullseye.pkr.hcl:
```diff
@@ -82,12 +70,12 @@ source "virtualbox-iso" "base-debian-amd64" {
-  ssh_password         = "${var.ssh_password}"
-  ssh_wait_timeout     = "30m"
+  ssh_wait_timeout     = "25m"
+  firmware             = "efi"
```
the 25m change was to speed up the manual test cycle. I will revert that once this work is ready to merge.

sdake commented:
Please note this PR is incomplete. The problem I am facing now, and I have a few changes not in this commit.

VirtualBox does not keep a record of efi nvvar variables. They are flushed on restart.

To work around this problem, the efi boot loader needs to be in a specific location. My work in progress has a change which is supposed to trigger an efi install that does not require efi nvvars.

Debian UEFI wiki has good information here. I have put about 10 references in the preseed.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,142 @@
 
 d-i partman-auto/choose_recipe select atomic
 d-i partman-partitioning/confirm_write_new_label boolean true
+d-i partman-basicfilesystems/no_swap boolean true
```
this no_swap line has no effect. A swap file is still generated.

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,142 @@
 
+# NB: https://wiki.debian.org/UEFI#Force_grub-efi_installation_to_the_removable_media_path
+d-i grub-installer/force-efi-extra-removable boolean true
+
```
This late command, currently, has three goals:
1. Configure GRUB's console to use serial
2. packer is in within the `sudo` group, however, set nopasswd for the packer user
3. Finally, delete `ifupdown` network configuration and use `systemd-networkd` configuration.

I am very wary of anything systemd related. I battled with `NetworkManager` for like 9 years of hard time at Red Hat. I am hopeful `systemd-networkd` learned from that experience - as we were all in the same tribe.

sdake commented on platform/golden/run.sh:
```diff
@@ -1,6 +1,9 @@
-SSH_PASSWORD=$(cat /dev/urandom | LC_ALL=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
-# Pass SSH password and build time to packer
-PACKER_LOG=1 packer build -var ssh_password=${SSH_PASSWORD} bullseye.pkr.hcl
+mkdir -p ${PWD}/packer_logs
```
I commented out the SSH password setting, as its hard to use, and still insecure by displaying the password in the shell.

rstarmer reviewed (approved):
Works, created image boots properly in BIOS mode.

sdake commented:
A boot with `cloud-hypervisor` functions according to goals. The network configuration has the appearance of functioning according to goals:

```
packer@debian:~$ sudo networkctl
IDX LINK TYPE     OPERATIONAL SETUP
  1 lo   loopback carrier     unmanaged
  2 ens3 ether    routable    configured

2 links listed.
```

sdake commented:
Rebuilding the apt index appears to function according to goals:
```
packer@debian:~$ sudo apt update
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://http.us.debian.org/debian bullseye InRelease
Hit:3 http://http.us.debian.org/debian bullseye-updates InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
```

Updating the apt package list appears to function according to goals:
```
packer@debian:~$ sudo apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

sdake commented:
The `dmesg` shows a valid kernel boot log. The results are `not` optimal, although they appear to function according to goals:
```
packer@debian:~$ sudo dmesg
[    0.000000] Linux version 5.10.0-23-amd64 (debian-kernel@lists.debian.org) (gcc-10 (Debian 10.2.1-6) 10.2.1 20210110, GNU ld (GNU Binutils for Debian) 2.35.2) #1 SMP Debian 5.10.179-1 (2023-05-12)
[    0.000000] Command line: BOOT_IMAGE=/boot/vmlinuz-5.10.0-23-amd64 root=UUID=c54a4aea-9578-40ba-8dfd-f5f39150a15f ro console=ttyS0,115200 quiet
[    0.000000] [Firmware Bug]: TSC doesn't count with P0 frequency!
[    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x004: 'AVX registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x200: 'Protection Keys User registers'
[    0.000000] x86/fpu: xstate_offset[2]:  576, xstate_sizes[2]:  256
[    0.000000] x86/fpu: xstate_offset[9]:  832, xstate_sizes[9]:    8
[    0.000000] x86/fpu: Enabled xstate features 0x207, context size is 840 bytes, using 'compacted' format.
[    0.000000] BIOS-provided physical RAM map:
[    0.000000] BIOS-e820: [mem 0x0000000000000000-0x000000000009ffff] usable
[    0.000000] BIOS-e820: [mem 0x0000000000100000-0x0000000000102fff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000103000-0x0000000000113fff] type 20
[    0.000000] BIOS-e820: [mem 0x0000000000114000-0x0000000000140fff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000141000-0x00000000bfffffff] usable
[    0.000000] BIOS-e820: [mem 0x0000000100000000-0x000000083fffffff] usable
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] efi: EFI v2.80 by H
[    0.000000] efi: ACPI 2.0=0xa0000
[    0.000000] secureboot: Secure boot disabled
[    0.000000] DMI not present or invalid.
[    0.000000] Hypervisor detected: KVM
[    0.000000] kvm-clock: Using msrs 4b564d01 and 4b564d00
[    0.000000] kvm-clock: cpu 0, msr 44de01001, primary cpu clock
[    0.000000] kvm-clock: using sched offset of 7454506962 cycles
[    0.000006] clocksource: kvm-clock: mask: 0xffffffffffffffff max_cycles: 0x1cd42e4dffb, max_idle_ns: 881590591483 ns
[    0.000010] tsc: Detected 2994.374 MHz processor
[    0.000151] e820: update [mem 0x00000000-0x00000fff] usable ==> reserved
[    0.000152] e820: remove [mem 0x000a0000-0x000fffff] usable
[    0.000158] last_pfn = 0x840000 max_arch_pfn = 0x400000000
[    0.000190] MTRR default type: write-back
[    0.000191] MTRR fixed ranges disabled:
[    0.000192]   00000-FFFFF uncachable
[    0.000192] MTRR variable ranges enabled:
[    0.000193]   0 disabled
[    0.000193]   1 disabled
[    0.000194]   2 disabled
[    0.000194]   3 disabled
[    0.000194]   4 disabled
[    0.000195]   5 disabled
[    0.000195]   6 disabled
[    0.000196]   7 disabled
[    0.000210] x86/PAT: Configuration [0-7]: WB  WC  UC- UC  WB  WP  UC- WT
[    0.000223] last_pfn = 0xc0000 max_arch_pfn = 0x400000000
[    0.000267] found SMP MP-table at [mem 0x000f0090-0x000f009f]
[    0.000379] Using GB pages for direct mapping
[    0.000561] RAMDISK: [mem 0x347c1000-0x363d7fff]
[    0.000578] ACPI: Early table checksum verification disabled
[    0.000605] ACPI: RSDP 0x00000000000A0000 000024 (v02 CLOUDH)
[    0.000611] ACPI: XSDT 0x00000000000A21C3 00003C (v01 CLOUDH CHXSDT   00000001 CLDH 00000000)
[    0.000620] ACPI: FACP 0x00000000000A1F31 000114 (v06 CLOUDH CHFACP   00000001 CLDH 00000000)
[    0.000624] ACPI: DSDT 0x00000000000A0024 001F0D (v06 CLOUDH CHDSDT   00000001 CLDH 00000000)
[    0.000626] ACPI: APIC 0x00000000000A2045 000142 (v05 CLOUDH CHMADT   00000001 CLDH 00000000)
[    0.000628] ACPI: MCFG 0x00000000000A2187 00003C (v01 CLOUDH CHMCFG   00000001 CLDH 00000000)
[    0.000630] ACPI: Reserving FACP table memory at [mem 0xa1f31-0xa2044]
[    0.000631] ACPI: Reserving DSDT table memory at [mem 0xa0024-0xa1f30]
[    0.000631] ACPI: Reserving APIC table memory at [mem 0xa2045-0xa2186]
[    0.000631] ACPI: Reserving MCFG table memory at [mem 0xa2187-0xa21c2]
[    0.000656] ACPI: Local APIC address 0xfee00000
[    0.000859] No NUMA configuration found
[    0.000861] Faking a node at [mem 0x0000000000000000-0x000000083fffffff]
[    0.000873] NODE_DATA(0) allocated [mem 0x83ffd6000-0x83fffffff]
[    0.001274] Zone ranges:
[    0.001277]   DMA      [mem 0x0000000000001000-0x0000000000ffffff]
[    0.001279]   DMA32    [mem 0x0000000001000000-0x00000000ffffffff]
[    0.001280]   Normal   [mem 0x0000000100000000-0x000000083fffffff]
[    0.001281]   Device   empty
[    0.001283] Movable zone start for each node
[    0.001284] Early memory node ranges
[    0.001286]   node   0: [mem 0x0000000000001000-0x000000000009ffff]
[    0.001286]   node   0: [mem 0x0000000000141000-0x00000000bfffffff]
[    0.001287]   node   0: [mem 0x0000000100000000-0x000000083fffffff]
[    0.001291] Initmem setup node 0 [mem 0x0000000000001000-0x000000083fffffff]
[    0.001292] On node 0 totalpages: 8388446
[    0.001295]   DMA zone: 64 pages used for memmap
[    0.001296]   DMA zone: 3396 pages reserved
[    0.001297]   DMA zone: 3934 pages, LIFO batch:0
[    0.001298]   DMA32 zone: 12224 pages used for memmap
[    0.001299]   DMA32 zone: 782336 pages, LIFO batch:63
[    0.001299]   Normal zone: 118784 pages used for memmap
[    0.001300]   Normal zone: 7602176 pages, LIFO batch:63
[    0.001311] On node 0, zone DMA: 1 pages in unavailable ranges
[    0.001454] On node 0, zone DMA: 161 pages in unavailable ranges
[    0.030084] ACPI: PM-Timer IO Port: 0x608
[    0.030086] ACPI: Local APIC address 0xfee00000
[    0.030144] IOAPIC[0]: apic_id 0, version 17, address 0xfec00000, GSI 0-23
[    0.030145] ACPI: INT_SRC_OVR (bus 0 bus_irq 4 global_irq 4 dfl dfl)
[    0.030149] Using ACPI (MADT) for SMP configuration information
[    0.030157] TSC deadline timer available
[    0.030163] smpboot: Allowing 32 CPUs, 0 hotplug CPUs
[    0.030179] kvm-guest: KVM setup pv remote TLB flush
[    0.030183] kvm-guest: setup PV sched yield
[    0.030206] PM: hibernation: Registered nosave memory: [mem 0x00000000-0x00000fff]
[    0.030207] PM: hibernation: Registered nosave memory: [mem 0x000a0000-0x000fffff]
[    0.030208] PM: hibernation: Registered nosave memory: [mem 0x00100000-0x00102fff]
[    0.030208] PM: hibernation: Registered nosave memory: [mem 0x00103000-0x00113fff]
[    0.030209] PM: hibernation: Registered nosave memory: [mem 0x00114000-0x00140fff]
[    0.030209] PM: hibernation: Registered nosave memory: [mem 0xc0000000-0xffffffff]
[    0.030211] [mem 0xc0000000-0xffffffff] available for PCI devices
[    0.030220] Booting paravirtualized kernel on KVM
[    0.030231] clocksource: refined-jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645519600211568 ns
[    0.033135] setup_percpu: NR_CPUS:8192 nr_cpumask_bits:32 nr_cpu_ids:32 nr_node_ids:1
[    0.037979] percpu: Embedded 63 pages/cpu s221184 r8192 d28672 u262144
[    0.037992] pcpu-alloc: s221184 r8192 d28672 u262144 alloc=1*2097152
[    0.037995] pcpu-alloc: [0] 00 01 02 03 04 05 06 07 [0] 08 09 10 11 12 13 14 15
[    0.038001] pcpu-alloc: [0] 16 17 18 19 20 21 22 23 [0] 24 25 26 27 28 29 30 31
[    0.038028] kvm-guest: stealtime: cpu 0, msr 81f635080
[    0.038030] kvm-guest: PV spinlocks enabled
[    0.038034] PV qspinlock hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.038058] Built 1 zonelists, mobility grouping on.  Total pages: 8253978
[    0.038059] Policy zone: Normal
[    0.038060] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-5.10.0-23-amd64 root=UUID=c54a4aea-9578-40ba-8dfd-f5f39150a15f ro console=ttyS0,115200 quiet
[    0.038152] printk: log_buf_len individual max cpu contribution: 4096 bytes
[    0.038152] printk: log_buf_len total cpu_extra contributions: 126976 bytes
[    0.038153] printk: log_buf_len min size: 131072 bytes
[    0.038805] printk: log_buf_len: 262144 bytes
[    0.038806] printk: early log buf free: 123880(94%)
[    0.056024] Dentry cache hash table entries: 4194304 (order: 13, 33554432 bytes, linear)
[    0.064628] Inode-cache hash table entries: 2097152 (order: 12, 16777216 bytes, linear)
[    0.064751] mem auto-init: stack:off, heap alloc:on, heap free:off
[    0.099535] Memory: 2919352K/33553784K available (12295K kernel code, 2536K rwdata, 7568K rodata, 2668K init, 17720K bss, 989048K reserved, 0K cma-reserved)
[    0.100126] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=32, Nodes=1
[    0.100167] ftrace: allocating 36577 entries in 143 pages
[    0.114821] ftrace: allocated 143 pages with 5 groups
[    0.115197] rcu: Hierarchical RCU implementation.
[    0.115201] rcu: 	RCU restricting CPUs from NR_CPUS=8192 to nr_cpu_ids=32.
[    0.115201] 	Rude variant of Tasks RCU enabled.
[    0.115202] 	Tracing variant of Tasks RCU enabled.
[    0.115202] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.115203] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=32
[    0.118381] NR_IRQS: 524544, nr_irqs: 664, preallocated irqs: 0
[    0.118452] random: crng init done
[    0.118484] Console: colour dummy device 80x25
[    0.118529] printk: console [ttyS0] enabled
[    0.118550] ACPI: Core revision 20200925
[    0.118584] APIC: Switch to symmetric I/O mode setup
[    0.118847] x2apic enabled
[    0.119889] Switched APIC routing to physical x2apic.
[    0.119890] kvm-guest: setup PV IPIs
[    0.119938] clocksource: tsc-early: mask: 0xffffffffffffffff max_cycles: 0x2b29828a32c, max_idle_ns: 440795270613 ns
[    0.119942] Calibrating delay loop (skipped) preset value.. 5988.74 BogoMIPS (lpj=11977496)
[    0.119944] pid_max: default: 32768 minimum: 301
[    0.122870] LSM: Security Framework initializing
[    0.122879] Yama: disabled by default; enable with sysctl kernel.yama.*
[    0.122925] AppArmor: AppArmor initialized
[    0.122926] TOMOYO Linux initialized
[    0.123222] Mount-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.123491] Mountpoint-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.123939] x86/cpu: User Mode Instruction Prevention (UMIP) activated
[    0.123939] Last level iTLB entries: 4KB 512, 2MB 512, 4MB 256
[    0.123939] Last level dTLB entries: 4KB 2048, 2MB 2048, 4MB 1024, 1GB 0
[    0.123939] Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
[    0.123939] Spectre V2 : Mitigation: Retpolines
[    0.123939] Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
[    0.123939] Spectre V2 : Spectre v2 / SpectreRSB : Filling RSB on VMEXIT
[    0.123939] Spectre V2 : Enabling Restricted Speculation for firmware calls
[    0.123939] Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
[    0.123939] Speculative Store Bypass: Mitigation: Speculative Store Bypass disabled via prctl and seccomp
[    0.123939] Freeing SMP alternatives memory: 32K
[    0.123939] smpboot: CPU0: AMD EPYC 7313 16-Core Processor (family: 0x19, model: 0x1, stepping: 0x1)
[    0.123939] Performance Events: Fam17h+ core perfctr, AMD PMU driver.
[    0.123939] ... version:                0
[    0.123939] ... bit width:              48
[    0.123939] ... generic registers:      6
[    0.123939] ... value mask:             0000ffffffffffff
[    0.123939] ... max period:             00007fffffffffff
[    0.123939] ... fixed-purpose events:   0
[    0.123939] ... event mask:             000000000000003f
[    0.123939] rcu: Hierarchical SRCU implementation.
[    0.123939] smp: Bringing up secondary CPUs ...
[    0.123939] x86: Booting SMP configuration:
[    0.123939] .... node  #0, CPUs:        #1
[    0.005505] kvm-clock: cpu 1, msr 44de01041, secondary cpu clock
[    0.123939] kvm-guest: stealtime: cpu 1, msr 81f675080
[    0.123939]   #2
[    0.005505] kvm-clock: cpu 2, msr 44de01081, secondary cpu clock
[    0.123939] kvm-guest: stealtime: cpu 2, msr 81f6b5080
[    0.124133]   #3
[    0.005505] kvm-clock: cpu 3, msr 44de010c1, secondary cpu clock
[    0.124576] kvm-guest: stealtime: cpu 3, msr 81f6f5080
[    0.124576]   #4
[    0.005505] kvm-clock: cpu 4, msr 44de01101, secondary cpu clock
[    0.125098] kvm-guest: stealtime: cpu 4, msr 81f735080
[    0.125098]   #5
[    0.005505] kvm-clock: cpu 5, msr 44de01141, secondary cpu clock
[    0.125098] kvm-guest: stealtime: cpu 5, msr 81f775080
[    0.125098]   #6
[    0.005505] kvm-clock: cpu 6, msr 44de01181, secondary cpu clock
[    0.125098] kvm-guest: stealtime: cpu 6, msr 81f7b5080
[    0.125098]   #7
[    0.005505] kvm-clock: cpu 7, msr 44de011c1, secondary cpu clock
[    0.128613] kvm-guest: stealtime: cpu 7, msr 81f7f5080
[    0.128613]   #8
[    0.005505] kvm-clock: cpu 8, msr 44de01201, secondary cpu clock
[    0.128613] kvm-guest: stealtime: cpu 8, msr 81f835080
[    0.128613]   #9
[    0.005505] kvm-clock: cpu 9, msr 44de01241, secondary cpu clock
[    0.128930] kvm-guest: stealtime: cpu 9, msr 81f875080
[    0.128930]  #10
[    0.005505] kvm-clock: cpu 10, msr 44de01281, secondary cpu clock
[    0.128930] kvm-guest: stealtime: cpu 10, msr 81f8b5080
[    0.128930]  #11
[    0.005505] kvm-clock: cpu 11, msr 44de012c1, secondary cpu clock
[    0.132042] kvm-guest: stealtime: cpu 11, msr 81f8f5080
[    0.132119]  #12
[    0.005505] kvm-clock: cpu 12, msr 44de01301, secondary cpu clock
[    0.132594] kvm-guest: stealtime: cpu 12, msr 81f935080
[    0.132594]  #13
[    0.005505] kvm-clock: cpu 13, msr 44de01341, secondary cpu clock
[    0.132887] kvm-guest: stealtime: cpu 13, msr 81f975080
[    0.132887]  #14
[    0.005505] kvm-clock: cpu 14, msr 44de01381, secondary cpu clock
[    0.132925] kvm-guest: stealtime: cpu 14, msr 81f9b5080
[    0.132925]  #15
[    0.005505] kvm-clock: cpu 15, msr 44de013c1, secondary cpu clock
[    0.132925] kvm-guest: stealtime: cpu 15, msr 81f9f5080
[    0.136112]  #16
[    0.005505] kvm-clock: cpu 16, msr 44de01401, secondary cpu clock
[    0.137052] kvm-guest: stealtime: cpu 16, msr 81fa35080
[    0.137052]  #17
[    0.005505] kvm-clock: cpu 17, msr 44de01441, secondary cpu clock
[    0.137052] kvm-guest: stealtime: cpu 17, msr 81fa75080
[    0.137052]  #18
[    0.005505] kvm-clock: cpu 18, msr 44de01481, secondary cpu clock
[    0.137052] kvm-guest: stealtime: cpu 18, msr 81fab5080
[    0.137052]  #19
[    0.005505] kvm-clock: cpu 19, msr 44de014c1, secondary cpu clock
[    0.139984] kvm-guest: stealtime: cpu 19, msr 81faf5080
[    0.140104]  #20
[    0.005505] kvm-clock: cpu 20, msr 44de01501, secondary cpu clock
[    0.140556] kvm-guest: stealtime: cpu 20, msr 81fb35080
[    0.140556]  #21
[    0.005505] kvm-clock: cpu 21, msr 44de01541, secondary cpu clock
[    0.141026] kvm-guest: stealtime: cpu 21, msr 81fb75080
[    0.141026]  #22
[    0.005505] kvm-clock: cpu 22, msr 44de01581, secondary cpu clock
[    0.141026] kvm-guest: stealtime: cpu 22, msr 81fbb5080
[    0.141026]  #23
[    0.005505] kvm-clock: cpu 23, msr 44de015c1, secondary cpu clock
[    0.141026] kvm-guest: stealtime: cpu 23, msr 81fbf5080
[    0.141026]  #24
[    0.005505] kvm-clock: cpu 24, msr 44de01601, secondary cpu clock
[    0.144669] kvm-guest: stealtime: cpu 24, msr 81fc35080
[    0.144669]  #25
[    0.005505] kvm-clock: cpu 25, msr 44de01641, secondary cpu clock
[    0.144669] kvm-guest: stealtime: cpu 25, msr 81fc75080
[    0.144669]  #26
[    0.005505] kvm-clock: cpu 26, msr 44de01681, secondary cpu clock
[    0.144669] kvm-guest: stealtime: cpu 26, msr 81fcb5080
[    0.144669]  #27
[    0.005505] kvm-clock: cpu 27, msr 44de016c1, secondary cpu clock
[    0.144945] kvm-guest: stealtime: cpu 27, msr 81fcf5080
[    0.144945]  #28
[    0.005505] kvm-clock: cpu 28, msr 44de01701, secondary cpu clock
[    0.144945] kvm-guest: stealtime: cpu 28, msr 81fd35080
[    0.148032]  #29
[    0.005505] kvm-clock: cpu 29, msr 44de01741, secondary cpu clock
[    0.148901] kvm-guest: stealtime: cpu 29, msr 81fd75080
[    0.148901]  #30
[    0.005505] kvm-clock: cpu 30, msr 44de01781, secondary cpu clock
[    0.148901] kvm-guest: stealtime: cpu 30, msr 81fdb5080
[    0.148901]  #31
[    0.005505] kvm-clock: cpu 31, msr 44de017c1, secondary cpu clock
[    0.148946] kvm-guest: stealtime: cpu 31, msr 81fdf5080
[    0.148946] smp: Brought up 1 node, 32 CPUs
[    0.148946] smpboot: Max logical packages: 1
[    0.148946] smpboot: Total of 32 processors activated (191639.93 BogoMIPS)
[    0.276639] node 0 deferred pages initialised in 116ms
[    0.280134] devtmpfs: initialized
[    0.280134] x86/mm: Memory block size: 128MB
[    0.280170] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.280207] futex hash table entries: 8192 (order: 7, 524288 bytes, linear)
[    0.280382] pinctrl core: initialized pinctrl subsystem
[    0.282353] NET: Registered protocol family 16
[    0.285183] DMA: preallocated 4096 KiB GFP_KERNEL pool for atomic allocations
[    0.287383] DMA: preallocated 4096 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.288672] DMA: preallocated 4096 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.288686] audit: initializing netlink subsys (disabled)
[    0.291981] audit: type=2000 audit(1687893394.291:1): state=initialized audit_enabled=0 res=1
[    0.292074] thermal_sys: Registered thermal governor 'fair_share'
[    0.292075] thermal_sys: Registered thermal governor 'bang_bang'
[    0.292077] thermal_sys: Registered thermal governor 'step_wise'
[    0.292077] thermal_sys: Registered thermal governor 'user_space'
[    0.292080] thermal_sys: Registered thermal governor 'power_allocator'
[    0.292111] cpuidle: using governor ladder
[    0.292111] cpuidle: using governor menu
[    0.292445] ACPI: bus type PCI registered
[    0.292452] acpiphp: ACPI Hot Plug PCI Controller Driver version: 0.5
[    0.292794] PCI: MMCONFIG for domain 0000 [bus 00-00] at [mem 0xe8000000-0xe80fffff] (base 0xe8000000)
[    0.292800] PCI: not using MMCONFIG
[    0.292812] PCI: Using configuration type 1 for base access
[    0.292812] PCI: Using configuration type 1 for extended access
[    0.296086] Kprobes globally optimized
[    0.296103] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.296103] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.388720] ACPI: Added _OSI(Module Device)
[    0.388720] ACPI: Added _OSI(Processor Device)
[    0.388720] ACPI: Added _OSI(3.0 _SCP Extensions)
[    0.388720] ACPI: Added _OSI(Processor Aggregator Device)
[    0.388720] ACPI: Added _OSI(Linux-Dell-Video)
[    0.388720] ACPI: Added _OSI(Linux-Lenovo-NV-HDMI-Audio)
[    0.388720] ACPI: Added _OSI(Linux-HPI-Hybrid-Graphics)
[    0.388720] ACPI: 1 ACPI AML tables successfully acquired and loaded
[    0.392654] ACPI: Interpreter enabled
[    0.392667] ACPI: (supports S0 S5)
[    0.392669] ACPI: Using IOAPIC for interrupt routing
[    0.392679] PCI: MMCONFIG for domain 0000 [bus 00-00] at [mem 0xe8000000-0xe80fffff] (base 0xe8000000)
[    0.392744] PCI: MMCONFIG at [mem 0xe8000000-0xe80fffff] reserved in ACPI motherboard resources
[    0.392760] PCI: Using host bridge windows from ACPI; if necessary, use "pci=nocrs" and report a bug
[    0.395161] ACPI: PCI Root Bridge [PCI0] (domain 0000 [bus 00])
[    0.395169] acpi PNP0A08:00: _OSC: OS supports [ExtendedConfig ASPM ClockPM Segments MSI HPX-Type3]
[    0.395172] acpi PNP0A08:00: _OSC failed (AE_NOT_FOUND); disabling ASPM
[    0.395305] acpiphp: Slot [0] registered
[    0.395325] acpiphp: Slot [1] registered
[    0.395335] acpiphp: Slot [2] registered
[    0.395343] acpiphp: Slot [3] registered
[    0.395352] acpiphp: Slot [4] registered
[    0.395360] acpiphp: Slot [5] registered
[    0.395368] acpiphp: Slot [6] registered
[    0.395378] acpiphp: Slot [7] registered
[    0.395386] acpiphp: Slot [8] registered
[    0.395394] acpiphp: Slot [9] registered
[    0.395402] acpiphp: Slot [10] registered
[    0.395413] acpiphp: Slot [11] registered
[    0.395421] acpiphp: Slot [12] registered
[    0.395429] acpiphp: Slot [13] registered
[    0.395439] acpiphp: Slot [14] registered
[    0.395447] acpiphp: Slot [15] registered
[    0.395455] acpiphp: Slot [16] registered
[    0.395463] acpiphp: Slot [17] registered
[    0.395471] acpiphp: Slot [18] registered
[    0.395478] acpiphp: Slot [19] registered
[    0.395488] acpiphp: Slot [20] registered
[    0.395495] acpiphp: Slot [21] registered
[    0.395503] acpiphp: Slot [22] registered
[    0.395514] acpiphp: Slot [23] registered
[    0.395524] acpiphp: Slot [24] registered
[    0.395532] acpiphp: Slot [25] registered
[    0.395541] acpiphp: Slot [26] registered
[    0.395549] acpiphp: Slot [27] registered
[    0.395558] acpiphp: Slot [28] registered
[    0.395570] acpiphp: Slot [29] registered
[    0.395583] acpiphp: Slot [30] registered
[    0.395591] acpiphp: Slot [31] registered
[    0.395610] PCI host bridge to bus 0000:00
[    0.395612] pci_bus 0000:00: root bus resource [mem 0xc0000000-0xe7ffffff window]
[    0.395613] pci_bus 0000:00: root bus resource [mem 0x840000000-0x7ff3fffffff window]
[    0.395614] pci_bus 0000:00: root bus resource [io  0x0000-0x0cf7 window]
[    0.395615] pci_bus 0000:00: root bus resource [io  0x0d00-0xffff window]
[    0.395615] pci_bus 0000:00: root bus resource [bus 00]
[    0.395674] pci 0000:00:00.0: [8086:0d57] type 00 class 0x060000
[    0.396030] pci 0000:00:01.0: [1af4:1043] type 00 class 0xffff00
[    0.396133] pci 0000:00:01.0: reg 0x10: [mem 0x7ff3ff80000-0x7ff3fffffff 64bit]
[    0.396647] pci 0000:00:02.0: [1af4:1042] type 00 class 0x018000
[    0.396696] pci 0000:00:02.0: reg 0x10: [mem 0xe7f80000-0xe7ffffff]
[    0.397183] pci 0000:00:03.0: [1af4:1041] type 00 class 0x020000
[    0.397245] pci 0000:00:03.0: reg 0x10: [mem 0x7ff3ff00000-0x7ff3ff7ffff 64bit]
[    0.397715] pci 0000:00:04.0: [1af4:1044] type 00 class 0xffff00
[    0.397776] pci 0000:00:04.0: reg 0x10: [mem 0x7ff3fe80000-0x7ff3fefffff 64bit]
[    0.398248] pci 0000:00:05.0: [1af4:1053] type 00 class 0xffff00
[    0.398317] pci 0000:00:05.0: reg 0x10: [mem 0x7ff3fe00000-0x7ff3fe7ffff 64bit]
[    0.399664] iommu: Default domain type: Translated
[    0.399990] vgaarb: loaded
[    0.400112] EDAC MC: Ver: 3.0.0
[    0.400112] Registered efivars operations
[    0.400165] NetLabel: Initializing
[    0.400165] NetLabel:  domain hash size = 128
[    0.400166] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.400196] NetLabel:  unlabeled traffic allowed by default
[    0.400199] PCI: Using ACPI for IRQ routing
[    0.400200] PCI: pci_cache_line_size set to 64 bytes
[    0.400519] clocksource: Switched to clocksource kvm-clock
[    0.422973] VFS: Disk quotas dquot_6.6.0
[    0.423003] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.423276] AppArmor: AppArmor Filesystem Enabled
[    0.423294] pnp: PnP ACPI init
[    0.423371] system 00:00: [mem 0xe8000000-0xe80fffff] has been reserved
[    0.423379] system 00:00: Plug and Play ACPI device, IDs PNP0c02 (active)
[    0.423400] pnp 00:01: Plug and Play ACPI device, IDs PNP0501 (active)
[    0.424216] pnp: PnP ACPI: found 2 devices
[    0.431610] clocksource: acpi_pm: mask: 0xffffff max_cycles: 0xffffff, max_idle_ns: 2085701024 ns
[    0.431669] NET: Registered protocol family 2
[    0.432858] IP idents hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.434897] tcp_listen_portaddr_hash hash table entries: 16384 (order: 6, 262144 bytes, linear)
[    0.436108] TCP established hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.436812] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes, linear)
[    0.436863] TCP: Hash tables configured (established 262144 bind 65536)
[    0.436911] UDP hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.436958] UDP-Lite hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.437072] NET: Registered protocol family 1
[    0.437082] NET: Registered protocol family 44
[    0.437100] pci_bus 0000:00: resource 4 [mem 0xc0000000-0xe7ffffff window]
[    0.437101] pci_bus 0000:00: resource 5 [mem 0x840000000-0x7ff3fffffff window]
[    0.437102] pci_bus 0000:00: resource 6 [io  0x0000-0x0cf7 window]
[    0.437103] pci_bus 0000:00: resource 7 [io  0x0d00-0xffff window]
[    0.437198] PCI: CLS 0 bytes, default 64
[    0.437265] Trying to unpack rootfs image as initramfs...
[    0.765986] Freeing initrd memory: 28764K
[    0.766005] PCI-DMA: Using software bounce buffering for IO (SWIOTLB)
[    0.766007] software IO TLB: mapped [mem 0x00000000bc000000-0x00000000c0000000] (64MB)
[    0.770441] clocksource: tsc: mask: 0xffffffffffffffff max_cycles: 0x2b29828a32c, max_idle_ns: 440795270613 ns
[    0.770855] clocksource: Switched to clocksource tsc
[    0.770933] platform rtc_cmos: registered platform RTC device (no PNP device found)
[    0.772224] Initialise system trusted keyrings
[    0.772237] Key type blacklist registered
[    0.772462] workingset: timestamp_bits=36 max_order=23 bucket_order=0
[    0.773353] zbud: loaded
[    0.774220] integrity: Platform Keyring initialized
[    0.774223] Key type asymmetric registered
[    0.774225] Asymmetric key parser 'x509' registered
[    0.774244] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 251)
[    0.774466] io scheduler mq-deadline registered
[    0.776063] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    0.777861] Serial: 8250/16550 driver, 4 ports, IRQ sharing enabled
[    0.778528] 00:01: ttyS0 at I/O 0x3f8 (irq = 24, base_baud = 115200) is a 16550A
[    0.779230] Linux agpgart interface v0.103
[    0.779256] AMD-Vi: AMD IOMMUv2 functionality not available on this system - This is not a bug.
[    0.787263] i8042: PNP: No PS/2 controller found.
[    0.787434] mousedev: PS/2 mouse device common for all mice
[    0.787516] rtc_cmos rtc_cmos: only 24-hr supported
[    0.789098] ledtrig-cpu: registered to indicate activity on CPUs
[    0.789253] NET: Registered protocol family 10
[    0.805334] Segment Routing with IPv6
[    0.805406] mip6: Mobile IPv6
[    0.805410] NET: Registered protocol family 17
[    0.805624] mpls_gso: MPLS GSO support
[    0.812935] IPI shorthand broadcast: enabled
[    0.812968] sched_clock: Marking stable (811351019, 1505663)->(853600041, -40743359)
[    0.813511] registered taskstats version 1
[    0.813519] Loading compiled-in X.509 certificates
[    0.856000] Loaded X.509 cert 'Debian Secure Boot CA: 6ccece7e4c6c0d1f6149f3dd27dfcc5cbb419ea1'
[    0.856034] Loaded X.509 cert 'Debian Secure Boot Signer 2022 - linux: 14011249c2675ea8e5148542202005810584b25f'
[    0.856993] zswap: loaded using pool lzo/zbud
[    0.858084] Key type .fscrypt registered
[    0.858085] Key type fscrypt-provisioning registered
[    0.858132] AppArmor: AppArmor sha1 policy hashing enabled
[    0.858350] integrity: Couldn't get size: 0x8000000000000003
[    0.858783] integrity: MODSIGN: Couldn't get UEFI db list
[    0.859270] integrity: Couldn't get size: 0x8000000000000003
[    0.859673] integrity: Couldn't get UEFI dbx list
[    0.863913] Freeing unused decrypted memory: 2036K
[    0.866795] Freeing unused kernel image (initmem) memory: 2668K
[    0.876037] Write protecting the kernel read-only data: 22528k
[    0.878959] Freeing unused kernel image (text/rodata gap) memory: 2040K
[    0.880469] Freeing unused kernel image (rodata/data gap) memory: 624K
[    0.934740] x86/mm: Checked W+X mappings: passed, no W+X pages found.
[    0.934750] Run /init as init process
[    0.934753]   with arguments:
[    0.934754]     /init
[    0.934755]   with environment:
[    0.934755]     HOME=/
[    0.934756]     TERM=linux
[    0.934757]     BOOT_IMAGE=/boot/vmlinuz-5.10.0-23-amd64
[    1.062312] virtio-pci 0000:00:01.0: enabling device (0000 -> 0002)
[    1.062687] virtio-pci 0000:00:02.0: enabling device (0000 -> 0002)
[    1.063548] virtio-pci 0000:00:03.0: enabling device (0000 -> 0002)
[    1.063877] virtio-pci 0000:00:04.0: enabling device (0000 -> 0002)
[    1.064256] virtio-pci 0000:00:05.0: enabling device (0000 -> 0002)
[    1.081242] virtio_blk virtio1: [vda] 134217728 512-byte logical blocks (68.7 GB/64.0 GiB)
[    1.081247] vda: detected capacity change from 0 to 68719476736
[    1.120716] virtio_net virtio2 ens3: renamed from eth0
[    1.141368]  vda: vda1 vda2 vda3
[    1.236230] PM: Image not found (code -22)
[    1.463392] SGI XFS with ACLs, security attributes, realtime, quota, no debug enabled
[    1.465299] XFS (vda2): Mounting V5 Filesystem
[    1.473104] XFS (vda2): Ending clean mount
[    1.502399] Not activating Mandatory Access Control as /sbin/tomoyo-init does not exist.
[    1.556331] systemd[1]: Inserted module 'autofs4'
[    1.577558] systemd[1]: systemd 247.3-7+deb11u2 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified)
[    1.577652] systemd[1]: Detected virtualization kvm.
[    1.577660] systemd[1]: Detected architecture x86-64.
[    1.578337] systemd[1]: Set hostname to <debian>.
[    1.745371] systemd[1]: Queued start job for default target Graphical Interface.
[    1.747541] systemd[1]: Created slice system-getty.slice.
[    1.747886] systemd[1]: Created slice system-modprobe.slice.
[    1.748393] systemd[1]: Created slice system-serial\x2dgetty.slice.
[    1.748626] systemd[1]: Created slice User and Session Slice.
[    1.748688] systemd[1]: Started Dispatch Password Requests to Console Directory Watch.
[    1.748720] systemd[1]: Started Forward Password Requests to Wall Directory Watch.
[    1.748895] systemd[1]: Set up automount Arbitrary Executable File Formats File System Automount Point.
[    1.748928] systemd[1]: Reached target Local Encrypted Volumes.
[    1.748954] systemd[1]: Reached target Paths.
[    1.748969] systemd[1]: Reached target Slices.
[    1.749543] systemd[1]: Listening on Syslog Socket.
[    1.749600] systemd[1]: Listening on initctl Compatibility Named Pipe.
[    1.749834] systemd[1]: Listening on Journal Audit Socket.
[    1.749916] systemd[1]: Listening on Journal Socket (/dev/log).
[    1.749996] systemd[1]: Listening on Journal Socket.
[    1.750129] systemd[1]: Listening on Network Service Netlink Socket.
[    1.750458] systemd[1]: Listening on udev Control Socket.
[    1.750532] systemd[1]: Listening on udev Kernel Socket.
[    1.751548] systemd[1]: Mounting Huge Pages File System...
[    1.752830] systemd[1]: Mounting POSIX Message Queue File System...
[    1.753970] systemd[1]: Mounting Kernel Debug File System...
[    1.755389] systemd[1]: Mounting Kernel Trace File System...
[    1.756893] systemd[1]: Starting Set the console keyboard layout...
[    1.757766] systemd[1]: Starting Create list of static device nodes for the current kernel...
[    1.758718] systemd[1]: Starting Load Kernel Module configfs...
[    1.759493] systemd[1]: Starting Load Kernel Module drm...
[    1.760657] systemd[1]: Starting Load Kernel Module fuse...
[    1.760905] systemd[1]: Condition check resulted in Set Up Additional Binary Formats being skipped.
[    1.763587] systemd[1]: Starting Journal Service...
[    1.764596] systemd[1]: Starting Load Kernel Modules...
[    1.765763] systemd[1]: Starting Remount Root and Kernel File Systems...
[    1.767162] systemd[1]: Starting Coldplug All udev Devices...
[    1.768625] systemd[1]: Mounted Huge Pages File System.
[    1.768738] systemd[1]: Mounted POSIX Message Queue File System.
[    1.768789] systemd[1]: Mounted Kernel Debug File System.
[    1.768831] systemd[1]: Mounted Kernel Trace File System.
[    1.769104] systemd[1]: Finished Create list of static device nodes for the current kernel.
[    1.769344] systemd[1]: modprobe@configfs.service: Succeeded.
[    1.769520] systemd[1]: Finished Load Kernel Module configfs.
[    1.770504] systemd[1]: Mounting Kernel Configuration File System...
[    1.772353] fuse: init (API version 7.32)
[    1.772744] systemd[1]: Mounted Kernel Configuration File System.
[    1.772883] systemd[1]: modprobe@fuse.service: Succeeded.
[    1.773094] systemd[1]: Finished Load Kernel Module fuse.
[    1.774035] systemd[1]: Mounting FUSE Control File System...
[    1.775336] xfs filesystem being remounted at / supports timestamps until 2038 (0x7fffffff)
[    1.776325] systemd[1]: Finished Remount Root and Kernel File Systems.
[    1.776454] systemd[1]: Mounted FUSE Control File System.
[    1.776657] systemd[1]: Condition check resulted in Rebuild Hardware Database being skipped.
[    1.776678] systemd[1]: Condition check resulted in Platform Persistent Storage Archival being skipped.
[    1.777554] systemd[1]: Starting Load/Save Random Seed...
[    1.778720] systemd[1]: Starting Create System Users...
[    1.783723] SCSI subsystem initialized
[    1.784164] systemd[1]: modprobe@drm.service: Succeeded.
[    1.784356] systemd[1]: Finished Load Kernel Module drm.
[    1.788834] Loading iSCSI transport class v2.0-870.
[    1.789093] systemd[1]: Finished Create System Users.
[    1.790184] systemd[1]: Finished Load/Save Random Seed.
[    1.790302] systemd[1]: Condition check resulted in First Boot Complete being skipped.
[    1.791094] systemd[1]: Starting Create Static Device Nodes in /dev...
[    1.795841] iscsi: registered transport (tcp)
[    1.799283] systemd[1]: Finished Create Static Device Nodes in /dev.
[    1.801152] systemd[1]: Starting Rule-based Manager for Device Events and Files...
[    1.817668] systemd[1]: Finished Set the console keyboard layout.
[    1.817728] systemd[1]: Reached target Local File Systems (Pre).
[    1.830317] systemd[1]: Finished Coldplug All udev Devices.
[    1.832414] systemd[1]: Starting Helper to synchronize boot up for ifupdown...
[    1.832593] systemd[1]: Started Rule-based Manager for Device Events and Files.
[    1.835214] systemd[1]: Started Journal Service.
[    1.838538] iscsi: registered transport (iser)
[    1.841719] systemd-journald[455]: Received client request to flush runtime journal.
[    1.856805] input: Power Button as /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0C:00/input/input0
[    1.856867] ACPI: Power Button [PWRB]
[    1.857470] input: PC Speaker as /devices/platform/pcspkr/input/input1
[    1.880588] NET: Registered protocol family 40
[    1.887750] cryptd: max_cpu_qlen set to 1000
[    1.931994] Adding 999420k swap on /dev/vda3.  Priority:-2 extents:1 across:999420k FS
[    1.932048] AVX2 version of gcm_enc/dec engaged.
[    1.932051] AES CTR mode by8 optimization enabled
[    2.050469] audit: type=1400 audit(1687893396.046:2): apparmor="STATUS" operation="profile_load" profile="unconfined" name="/usr/bin/man" pid=661 comm="apparmor_parser"
[    2.050476] audit: type=1400 audit(1687893396.046:3): apparmor="STATUS" operation="profile_load" profile="unconfined" name="man_filter" pid=661 comm="apparmor_parser"
[    2.050488] audit: type=1400 audit(1687893396.046:4): apparmor="STATUS" operation="profile_load" profile="unconfined" name="man_groff" pid=661 comm="apparmor_parser"
[    2.050500] audit: type=1400 audit(1687893396.046:5): apparmor="STATUS" operation="profile_load" profile="unconfined" name="lsb_release" pid=659 comm="apparmor_parser"
[    2.052608] audit: type=1400 audit(1687893396.050:6): apparmor="STATUS" operation="profile_load" profile="unconfined" name="nvidia_modprobe" pid=660 comm="apparmor_parser"
[    2.052614] audit: type=1400 audit(1687893396.050:7): apparmor="STATUS" operation="profile_load" profile="unconfined" name="nvidia_modprobe//kmod" pid=660 comm="apparmor_parser"
```

sdake commented:
@rstarmer you mentioned above this works in BIOS mode. The built image may work in bios mode, although it definitely works with efi firmware, as seen from the kernel dmesg boot log.

Would you clarify if you want an image that boots in both bios and efi?

sdake commented on platform/golden/cfg/preseed-debian-11.cfg:
```diff
@@ -1,95 +1,142 @@
 
-#######################
-## finish
+d-i debian-installer/add-kernel-opts console=tty1 console=ttyS0,115200
```
I am considering changing this in a follow-up PR such that kernel options are added in /etc/default/grub.d/option.cfg, rather than /etc/default/grub as this `di` change does.

rstarmer replied:
Note that the grub docs (https://www.gnu.org/software/grub/manual/grub/grub.html#Simple-configuration) point to simple changes like adding a console configuration, are valid in /etc/default/grub, so this doesn't _have_ to be changed unless there is some other requirement that drives this.

sdake commented on platform/golden/bullseye.pkr.hcl:
```diff
@@ -40,38 +40,27 @@ variable "iso_checksum_type" {
-  ]
+  # The Debian boot screen is well documented in 5.1.7 The Boot Screen
+  # https://www.debian.org/releases/stable/i386/ch05s01.en.html#boot-screen
+  boot_command = ["<down>e<down><down><down><end>priority=critical auto=true fb=false preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/${var.preseed_file}<leftCtrlOn>x<leftCtrlOff>"]
```
Robert had provided a review (by Discord) explaining the problem with the network interface mapping. His suggestion was to change this to the following:
```
+  boot_command = ["<down>e<down><down><down><end>priority=critical auto=true preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/${var.preseed_file} net.ifnames=0 biosdevname=0<leftCtrlOn>x<leftCtrlOff>"]

```
Unfortunately, I didn't understand the problem until I re-discovered the problem. I may have understood without having to rediscover the problem if the problem was left in the review. Whats great is that we are learning as a team, together!

sdake replied:
Also, I am not sure where the `biosdevname=0` cargo copy cult came from, however, this kernel option only applies to Dell systems that have the `biosdevname` package installed [1][2].

[1] https://linux.dell.com/files/biosdevname/
[2] https://unix.stackexchange.com/questions/637295/linux-kernel-parameters-what-is-the-difference-of-net-ifnames-0-and-biosdevname#:~:text=ifnames%20tells%20systemd%20to%20create,you%20may%20guess%20it%2C%20biosdevname%20.

rstarmer replied:
FYI, it came from Copilot/ChatGPT Q&A.

rstarmer reviewed (approved):
LGTM, works for BIOS based boot.
